### PR TITLE
Creates a generic Maze Drawer class

### PR DESCRIPTION
**Pull request by Hamms**
And refactors BeeItemDrawer and DirtDrawer to extend it.

Previously, BeeItemDrawer inherited from DirtDrawer, although DirtDrawer
contained a lot of functionality that BeeItemDrawer intentionally
discarded. This refactor plus ES6 class implementation makes their
shared vs individual functionality a bit cleaner, and will also make it
easier to add a new Drawer for the new WIP Collector Level type.

### CODE REVIEW

**Comment by Hamms** [comment]
I would highly recommend reviewing this PR with [whitespace ignored](https://github.com/code-dot-org/code-dot-org/pull/9224/files?w=1)

**Comment by islemaster** [comment]
This LGTM.  I'm a little surprised this change adds more code than it removes :) but it's a nice refactor. :+1:

**Comment by islemaster** [comment]
Looks like some tests depended on something you removed?

File: apps/src/maze/beeItemDrawer.js [hunk 1]
```diff
@@ -1,302 +1,301 @@
...
-require('../utils'); // Provides Function.prototype.inherits
-
-var cellId = require('./mazeUtils').cellId;
-
-var SVG_NS = require('../constants').SVG_NS;
-var SQUARE_SIZE = 50;
+import Drawer from './drawer';
+const SQUARE_SIZE = 50;
+let SVG_NS = require('../constants').SVG_NS;
+let cellId = require('./mazeUtils').cellId;
```
Comment (joshlory) [inline apps/src/maze/beeItemDrawer.js]: `const` or `import`?

**Comment by Hamms** [comment]
PTA(quick)L

File: apps/src/maze/dirtDrawer.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ export default class DirtDrawer extends Drawer {
    * @param {number} val
    * @return {number}
    */
-  spriteIndexForDirt_(val) {
+  static spriteIndexForDirt_(val) {
```
Comment (islemaster) [inline apps/src/maze/dirtDrawer.js]: Nice... I think this is the first time I've seen someone use the `static` keyword in JavaScript.

**Comment by islemaster** [comment]
LGTM

File: apps/src/maze/dirtDrawer.js (hunk 2 above)
Comment (Hamms) [inline apps/src/maze/dirtDrawer.js]: It's exciting to see it added to the language! JavaScript feels like it's growing up ... :joy:

**Pull request merged**